### PR TITLE
fix: Ensure unmaskTextSelector is used for masked attributes

### PR DESCRIPTION
**Pull request by mydea**
Closes https://github.com/getsentry/rrweb/issues/79

### CODE REVIEW

File: packages/rrweb-snapshot/src/snapshot.ts [hunk 1]
```diff
@@ -266,20 +267,34 @@ export function transformAttribute(
...
   }
 
   return value;
 }
 
+function _shouldMaskAttribute(
+  element: HTMLElement,
+  name: string,
+  tagName: string,
+  { unmaskTextSelector }: { unmaskTextSelector: string | undefined | null },
```
Comment (billyvg) [inline packages/rrweb-snapshot/src/snapshot.ts:282]: let's make this a single obj param?

File: packages/rrweb-snapshot/src/snapshot.ts [hunk 2]
```diff
@@ -266,20 +267,34 @@ export function transformAttribute(
...
   ) {
     return maskTextFn ? maskTextFn(value) : defaultMaskFn(value);
   }
 
   return value;
 }
 
+function _shouldMaskAttribute(
+  element: HTMLElement,
+  name: string,
```
Comment (billyvg) [inline packages/rrweb-snapshot/src/snapshot.ts:280]: Maybe `attribute` or `attributeName` to distinguish it a bit from `tagName`:


```suggestion
  attribute: string,
```

File: packages/rrweb/test/__snapshots__/integration.test.ts.snap [hunk 3]
```diff
@@ -4097,6 +4097,366 @@ exports[`record integration tests can use maskInputOptions to configure which ty
...
+                        },
+                        "childNodes": [
+                          {
+                            "type": 3,
+                            "textContent": "\\n        Test content\\n      ",
+                            "id": 21
+                          }
+                        ],
+                        "id": 20
+                      },
```
Comment (billyvg) [inline packages/rrweb/test/__snapshots__/integration.test.ts.snap:4254]: LGTM

File: packages/rrweb-snapshot/src/snapshot.ts (hunk 1 above)
Comment (mydea) [reply packages/rrweb-snapshot/src/snapshot.ts]: done!

**Review by billyvg — APPROVED** [review]

**Pull request merged**